prow/cmd/deck/user.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"k8s.io/test-infra/prow/userdashboard"
+)
+
+type userAgent struct {
```

Comment:
This object could use some comments about its purpose.
It's also not clear to me why we use a pointer and mutex here. It seems like we should have no problem handling many users at once if we handled this a bit differently..

Reply:
Yeah, the mutex was my mistake. Beside removing the mutex, is there any thing else can be done to handle requests properly?